task.ps1

```
@@ -1,30 +1,89 @@
-$location = "uksouth"
-$resourceGroupName = "mate-azure-task-16"
+param (
+    [string]$resourceGroupName = "mate-resources",
+    [string]$location = "East US",
+    [string]$vnetName = "mate-vnet"
+)
 
-$virtualNetworkName = "todoapp"
-$vnetAddressPrefix = "10.20.30.0/24"
+# Сабнети
 $webSubnetName = "webservers"
-$webSubnetIpRange = "10.20.30.0/26"
 $dbSubnetName = "database"
-$dbSubnetIpRange = "10.20.30.64/26"
-$mngSubnetName = "management"
-$mngSubnetIpRange = "10.20.30.128/26"
+$mgmtSubnetName = "management"
 
+# NSG
+$webNsgName = "webservers"
+$dbNsgName = "database"
+$mgmtNsgName = "management"
 
-Write-Host "Creating a resource group $resourceGroupName ..."
-New-AzResourceGroup -Name $resourceGroupName -Location $location
+# ===== VNET =====
+$vnet = Get-AzVirtualNetwork -Name $vnetName -ResourceGroupName $resourceGroupName -ErrorAction SilentlyContinue
+if (-not $vnet) {
+    Write-Host "Creating VNet..."
+    $vnet = New-AzVirtualNetwork -Name $vnetName -ResourceGroupName $resourceGroupName -Location $location -AddressPrefix "10.0.0.0/16"
+    Add-AzVirtualNetworkSubnetConfig -Name $webSubnetName -AddressPrefix "10.0.1.0/24" -VirtualNetwork $vnet | Out-Null
+    Add-AzVirtualNetworkSubnetConfig -Name $dbSubnetName -AddressPrefix "10.0.2.0/24" -VirtualNetwork $vnet | Out-Null
+    Add-AzVirtualNetworkSubnetConfig -Name $mgmtSubnetName -AddressPrefix "10.0.3.0/24" -VirtualNetwork $vnet | Out-Null
+    $vnet | Set-AzVirtualNetwork | Out-Null
+} else {
+    Write-Host "VNet already exists, using existing one."
```

When the VNet exists the script only prints "VNet already exists, using existing one." but does not ensure the three expected subnets exist. If the existing VNet is missing any of the webservers, database, or management subnets the later NSG association will silently skip them. Add logic here to check for and create any missing subnets (or fail with a clear message) so the script is robust and idempotent across existing VNets.